documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,473 @@
+        default: "30"
+      - name: "configuration"
+        in: body
+        schema:
+          type: object
+          items:
+            $ref: "#/definitions/EventPipeConfiguration"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /trace/{uid}:
+    post:
+      tags:
+      - "traces"
+      operationId: "posttrace"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "duration"
+        in: "query"
+        description: "Duration of the trace"
+        required: false
+        type: "integer"
+        default: "30"
+      - name: "uid"
+        in: "path"
+        description: "ID of the process to collect a process dump"
+        required: true
+        type: "string"
+      - name: "configuration"
+        in: body
+        schema:
+          type: object
+          items:
+            $ref: "#/definitions/EventPipeConfiguration"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /logs:
+    get:
+      tags:
+      - "logs"
+      operationId: "getdefaultlogs"
+      produces:
+      # Newline-delimited JSON- http://www.ietf.org/rfc/rfc4288.txt
+      - "application/x-ndjson"
+      - "text/event-stream"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /logs/{uid}:
+    get:
+      tags:
+      - "logs"
+      operationId: "getlogs"
+      produces:
+      - "application/x-ndjson"
+      - "text/event-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+    post:
+      tags:
+      - "logs"
+      operationId: "postlogs"
+      produces:
+      - "application/x-ndjson"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process"
+        required: true
+        type: "string"
+      - name: "configuration"
+        in: "body"
+        schema:
+          type: "array"
+          items:
+            $ref: "#/definitions/LogFilter"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+            
+  /metrics:
```

Comment:
Is this endpoint supposed to give back a stream of metrics? Or does it return a snapshot of the default metrics at a given point in time?

Reply:
It's a snapshot. We could add a streaming endpoint in addition to the Prometheus exposition format but it didn't look like there's any desire for that.